### PR TITLE
Custom disk cache path

### PR DESCRIPTION
**Pull request by LukeDurrant**

### CODE REVIEW

**Comment by mythodeia** [comment]
thanks for the contribution but there is [this](https://github.com/rs/SDWebImage/pull/1041) and [this](https://github.com/rs/SDWebImage/pull/1093)

**Comment by LukeDurrant** [comment]
Yeah thats not the same.
That doesn't allow you to make the directory not in the NSCachesDirectory.
Reason for moving this is extensions.

**Comment by mythodeia** [comment]
having NSCachesDirectory as a fallback is indeed a nice touch. i will reopen so everyone can see and use it if needed.

File: SDWebImage/SDImageCache.m [hunk 1]
```diff
@@ -101,7 +106,15 @@ - (id)initWithNamespace:(NSString *)ns {
         _memCache.name = fullNamespace;
 
         // Init the disk cache
-        _diskCachePath = [self makeDiskCachePath:fullNamespace];
+        if(directory!=nil)
```
Comment (bpoplauschi) [inline SDWebImage/SDImageCache.m]: please match our coding style

```
if (...) {
   ...
} else {
   ...
}
```

**Comment by sawijaya** [comment]
after i using custom disk cache path, the method clearDisk() not working for clear disk cache.
here is my code

let caches: [AnyObject]? = BOPreferences.sharedInstance.caches.allCaches()
            if let caches = caches {
                for(_, value) in caches.enumerate() {
                    var cache: [String: AnyObject] = value as! [String: AnyObject]
                    let namespace: String = cache["namespace"] as! String
                    let directory: String = cache["diskCacheDirectory"] as! String
                    let sdImageCache: SDImageCache = SDImageCache(namespace: namespace, diskCacheDirectory: directory) // => custom disk patch cache
                    sdImageCache.clearDisk() // ===> after call this not working (my cache file still on there)
                }
            }

anything wrong with my code for create custom cache and clear that custom cache?
why it's not working for clearDisk()?

**Pull request merged**